api/v1/kustomization_types.go

```diff
@@ -95,6 +99,13 @@ type KustomizationSpec struct {
 	// +required
 	Prune bool `json:"prune"`
 
+	// DeletionPolicy can be used to control garbage collection when this
+	// Kustomization is deleted. Will mirror value of Prune if unspecified.
+	// Valid values are ('MirrorPrune' (default), 'Delete')
+	// +kubebuilder:validation:Enum=MirrorPrune;Delete;Orphan
+	// +optional
+	DeletionPolicy *string `json:"deletionPolicy,omitempty"`
```

Comment:
```suggestion
	DeletionPolicy string `json:"deletionPolicy,omitempty"`
```

No need to set a pointer to a string.